src/malli/core.cljc

```diff
@@ -555,13 +555,13 @@
         (-intercepting this-transformer
                        (if (= :decode method)
                          (fn [x]
-                           (key (reduce-kv
-                                 (fn [acc i transformer]
-                                   (let [x* (transformer (val acc))]
-                                     (if ((nth validators i) x*)
-                                       (reduced (miu/-tagged x* nil))
-                                       (miu/-tagged (or (key acc) x*) x))))
-                                 (miu/-tagged nil x) transformers)))
+                           (reduce-kv
+                            (fn [acc i transformer]
+                              (let [x* (transformer x)]
+                                (if ((nth validators i) x*)
+                                  (reduced x*)
+                                  (or acc x*))))
```

Comment:
What about when the first `transformer` returns a `nil`? Could that happen?

I'm thinking of something like `(m/decode [:maybe :int] false false-to-nil-transformer) ==> nil`

Or perhaps even just `(m/decode [:or nil :int] nil trans)`?

Reply:
I thought it wouldn't matter, as `[:or nil :int]` with `nil` value succeeds and doesn't use the fallback but here's a case where is fails:

```clojure
(m/decode
 [:or
  [:string {:decode/remove-secrets (constantly nil)}]
  :keyword]
 "SECRET!"
 (mt/transformer {:name :remove-secrets}))
; => "SECRET!"
```

Fixed it so that value of the first branch is always used as fallback, regardless of value.